crates/rome_js_analyze/src/semantic_analyzers/nursery/use_iframe_title.rs

```
@@ -0,0 +1,150 @@
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe></iframe>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe {...props} />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title="" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={""} />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={<span className={"token string"}></span>}></iframe>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={undefined} />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={false} />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={true} />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    ///     <iframe title={42} />
+    /// ```
+    ///
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    ///     <>
+    ///         <iframe title="This is a unique title" />
+    ///         <iframe title={uniqueTitle} />
+    ///     </>
+    /// ```
+    ///
+    /// ## Accessibility guidelines
+    ///
+    /// [WCAG 2.4.1](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks)
+    /// [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+    ///
+    pub(crate) UseIframeTitle {
+        version: "12.0.0",
+        name: "useIframeTitle",
+        recommended: true,
+    }
+}
+
+pub(crate) struct UseIframeTitleState {
+    node: AnyJsxElement,
+}
+
+impl Rule for UseIframeTitle {
+    type Query = Ast<AnyJsxElement>;
+    type State = UseIframeTitleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let is_iframe = matches!(node.name_value_token()?.text_trimmed(), "iframe");
+        if !is_iframe {
+            return None;
+        }
+
+        if node.attributes().is_empty() {
```

If the diagnostic message is the same in both cases you can merge them like:

```rust
let Some(title_attribute) = node.find_attribute_by_name("title") else {
    return Some(UseIframeTitleState { node: node.clone() });
}
let attribute_value = title_attribute.initializer()?.value().ok()?;

match attribute_value  {
    JsxString(str) => is_valid_string(str),
    JsxExpressionAttributeValue(expr) => is_valid_expression(expr),
    AnyJsxTag => return Some(UseIframeTitleState { node: node.clone() });
}
```

If `node.attributes().is_empty()`, `find_attribute_by_name` will fail fast enough already.

---

fixed at [`e26c383` (#4067)](https://github.com/rome/tools/pull/4067/commits/e26c383d26a164c0f94c271b88deb886845ee226)